### ISSUE TITLE
NVDA stops reading text in the editor after using find functionality

### ISSUE THREAD

**Issue by venkateshpotluri** [microsoft/vscode#90788]
Issue Type: <b>Bug</b>

1. open a file. (tested with .tex files)
2. find for some text.
3. exit from the find dialog box by pressing escape.
4. try reading through the text using up-down arrow keys. NVDA announces "blank" for all lines of text.
5. press and hold the up arrow key to go to the top of the document. Now try reading the text using the down arrow key, and NVDA reads text as expected.
6. reload window with extensions disabled.
7. repeat steps 2-5.

Tested using NVDA2019.3.1.
## expected behavior:
NVDA should read text when arrow keys are pressed after closing the find dialog without having to go the top of the document. Here is the test file I used to reproduce this issue: [NVDAVSCodeBugTest.txt](https://github.com/microsoft/vscode/files/4233970/NVDAVSCodeBugTest.txt)

VS Code version: Code 1.42.1 (c47d83b293181d9be64f27ff093689e8e7aed054, 2020-02-11T14:45:59.656Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.78GB (8.30GB free)|
|Process Argv|.|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
spellright|ban|3.0.50
latex-workshop|Jam|8.7.2

</details>

**Comment by isidorn** [comment]
Thanks for filling this issue. 
I can not reproduce this. Can you reproduce using Visual Studio Code Insiders https://code.visualstudio.com/insiders/

**Comment by venkateshpotluri** [comment]
Hi,
I have tried with the insider build and notice similar behavior, but occurs intermittently. You could also potentially reproduce this issue by going to the bottom of the file by pressing control+end and then navigating up using the up arrow key. Note that the contents of my file are more than the vertical space of the screen. That is, a user would have to scroll to get to the bottom of the file. Unfortunately, I am unable to share the contents of the file, or the file publicly where this occurs else I would have been happy to record a video and would have added it to this issue. Please let me know if I can help in any way.

**Comment by isidorn** [comment]
Ok, I will try to repro again next week. Thus assigning to March when I plan to investigate deeper.
Potentially related https://github.com/nvaccess/nvda/issues/8781

**Comment by venkateshpotluri** [comment]
[This issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1001591&q=nvda%20does%20not%20read&can=2) might be the cause. I generated a file with > 1,000 'x's and NVDA stops reading the characters from line 125 (attaching the file to the issue for your reference). Also, I wanted to note that I was able to use VSCode fine with large files until recently (prior to the the January2020 update). When I noticed that NVDA was no longer announcing text in the TeX file, I updated NVDA to the latest version assuming that might fix it. Perhaps this is a regression of some sort?

Hope this helps.

**Comment by isidorn** [comment]
@venkateshpotluri this is very good feedback, thanks for providing more details.
Unfortunetly you did not attach the file. Can you share it so I also try it out with NVDA?

I might be that one issue is bothering you and I suspect this is related to https://github.com/microsoft/vscode/issues/89717 which boils down to the Chrome issue which you linked.

Why does this issue show now: because in the january milestone we changed our strategy, previously we were putting only 10 lines in the `contentArea` which the ScreenReader could reach. Now we put 160 lines. Idealy we would just put the whole document but due to performance issues on the Chrome side we can not.
If the actual problem is not line based (how I originaly thought) but character based than we should also decide on the line count based on the number of character. I will be able to determine that once you share your file for which content is not read out.

Thanks a lot

**Comment by venkateshpotluri** [comment]
Oh sorry. The markdown didn't format it well. Here is the link: [NVDAVSCodeBugTest.txt](https://github.com/microsoft/vscode/files/4233970/NVDAVSCodeBugTest.txt)

**Comment by isidorn** [comment]
Thanks. I'll continue the investigation next week, as we are now wrapping up our release and are focused on testing.

**Comment by isidorn** [comment]
@venkateshpotluri I acknolwedge the issue, and you are 100% right, https://bugs.chromium.org/p/chromium/issues/detail?id=1001591&q=nvda%20does%20not%20read&can=2 is the root cause of this issue.
Since we can not properly fix it, we can only heuristicly take the number of lines that we can put in the text area such that the limit of 10000 is not hit.
We were putting 160 lines and now I changed this that we put only 100 lines when a screen reader is detected.
THis is a poor workaround for a big issue in chrome, however we currently do not have a better solution.

I am open for ideas. But with the fix I pushed now the example you provided works just fine.

Please let us know if you hit more issues with VS Code and feel free to pine me @isidorn on those issues. Thanks

**Comment by venkateshpotluri** [comment]
Thanks for fixing this and updating! can I test this in the latest insider build? If I understand correct, the current fix may work for the file I attached to this issue, but a different file (with longer length) would reproduce this issue. Is this correct? If yes, what are your thoughts on making this value configurable? That way, users could potentially set the appropriate project specific value for projects in the .vscode/settings.json file depending on project size. I am not familiar with the code base so not sure how feasible this would be.

**Comment by isidorn** [comment]
@venkateshpotluri you can try it out in vscode insiders in a couple of days. Probably from Wednesday!
Yes it is correct what you just said and the value is actually configurable already, it is called 	`"editor.accessibilityPageSize": 
Set it to a lower number than 100 if you see issues.

**Comment by isidorn** [comment]
@venkateshpotluri did you get a chance to try this out? Asking because we would like to verify this issue if it is fixed for you. Thanks!
